http3/response_writer.go

```diff
@@ -232,6 +235,79 @@ func (w *responseWriter) Flush() {
 	}
 }
 
+func (w *responseWriter) promoteTrailer() {
+	for k, vv := range w.header {
+		if k == "Trailer" {
+			for _, val := range vv {
+				w.declareTrailer(val)
+			}
+		}
+		if !strings.HasPrefix(k, http2.TrailerPrefix) {
+			continue
+		}
+		trailerKey := strings.TrimPrefix(k, http2.TrailerPrefix)
+		w.declareTrailer(trailerKey)
+		w.header[http.CanonicalHeaderKey(trailerKey)] = vv
+	}
+}
+
+func strSliceContains(ss []string, s string) bool {
```

Comment:
`slices.Contains`?

Reply:
I switched the implementation around to use a map instead! But if it's kept a list, obviously `slices.Contains` is the better method for this